From a05db08b3328865028177ac9f2bba91e6445ca34 Mon Sep 17 00:00:00 2001
From: Jason Dere <jdere@hortonworks.com>
Date: Wed, 29 Jul 2015 14:26:16 -0700
Subject: [PATCH 0410/1164] CDH-34913: YHIVE-11143: Tests
 udf_from_utc_timestamp.q/udf_to_utc_timestamp.q
 do not work with updated Java timezone
 information (Jason Dere, reviewed by Alex
 Pivovarov)

Change-Id: I8413882674c0dcefb5615945a68613dd8c704e77
---
 .../clientpositive/udf_from_utc_timestamp.q        |   30 ++++-----
 .../queries/clientpositive/udf_to_utc_timestamp.q  |   30 ++++-----
 .../clientpositive/udf_from_utc_timestamp.q.out    |   66 ++++++++++----------
 .../clientpositive/udf_to_utc_timestamp.q.out      |   66 ++++++++++----------
 4 files changed, 96 insertions(+), 96 deletions(-)

diff --git a/ql/src/test/queries/clientpositive/udf_from_utc_timestamp.q b/ql/src/test/queries/clientpositive/udf_from_utc_timestamp.q
index de98507..b113eec 100644
--- a/ql/src/test/queries/clientpositive/udf_from_utc_timestamp.q
+++ b/ql/src/test/queries/clientpositive/udf_from_utc_timestamp.q
@@ -1,24 +1,24 @@
 DESCRIBE FUNCTION from_utc_timestamp;
 DESC FUNCTION EXTENDED from_utc_timestamp;
 
-explain select from_utc_timestamp('2015-02-11 10:30:00', 'PST');
+explain select from_utc_timestamp('2012-02-11 10:30:00', 'PST');
 
 select
-from_utc_timestamp('2015-02-11 04:30:00', 'PST'),
-from_utc_timestamp('2015-02-11 04:30:00', 'Europe/Moscow'),
-from_utc_timestamp('2015-02-11 04:30:00', 'GMT+8'),
-from_utc_timestamp('2015-02-11 04:30:00', 'GMT'),
-from_utc_timestamp('2015-02-11 04:30:00', ''),
-from_utc_timestamp('2015-02-11 04:30:00', '---'),
+from_utc_timestamp('2012-02-11 04:30:00', 'PST'),
+from_utc_timestamp('2012-02-11 04:30:00', 'Europe/Moscow'),
+from_utc_timestamp('2012-02-11 04:30:00', 'GMT+8'),
+from_utc_timestamp('2012-02-11 04:30:00', 'GMT'),
+from_utc_timestamp('2012-02-11 04:30:00', ''),
+from_utc_timestamp('2012-02-11 04:30:00', '---'),
 from_utc_timestamp(cast(null as string), 'PST'),
-from_utc_timestamp('2015-02-11 04:30:00', cast(null as string));
+from_utc_timestamp('2012-02-11 04:30:00', cast(null as string));
 
 select
-from_utc_timestamp(cast('2015-02-11 04:30:00' as timestamp), 'PST'),
-from_utc_timestamp(cast('2015-02-11 04:30:00' as timestamp), 'Europe/Moscow'),
-from_utc_timestamp(cast('2015-02-11 04:30:00' as timestamp), 'GMT+8'),
-from_utc_timestamp(cast('2015-02-11 04:30:00' as timestamp), 'GMT'),
-from_utc_timestamp(cast('2015-02-11 04:30:00' as timestamp), ''),
-from_utc_timestamp(cast('2015-02-11 04:30:00' as timestamp), '---'),
+from_utc_timestamp(cast('2012-02-11 04:30:00' as timestamp), 'PST'),
+from_utc_timestamp(cast('2012-02-11 04:30:00' as timestamp), 'Europe/Moscow'),
+from_utc_timestamp(cast('2012-02-11 04:30:00' as timestamp), 'GMT+8'),
+from_utc_timestamp(cast('2012-02-11 04:30:00' as timestamp), 'GMT'),
+from_utc_timestamp(cast('2012-02-11 04:30:00' as timestamp), ''),
+from_utc_timestamp(cast('2012-02-11 04:30:00' as timestamp), '---'),
 from_utc_timestamp(cast(null as timestamp), 'PST'),
-from_utc_timestamp(cast('2015-02-11 04:30:00' as timestamp), cast(null as string));
\ No newline at end of file
+from_utc_timestamp(cast('2012-02-11 04:30:00' as timestamp), cast(null as string));
diff --git a/ql/src/test/queries/clientpositive/udf_to_utc_timestamp.q b/ql/src/test/queries/clientpositive/udf_to_utc_timestamp.q
index fe0b647..cca6d7d 100644
--- a/ql/src/test/queries/clientpositive/udf_to_utc_timestamp.q
+++ b/ql/src/test/queries/clientpositive/udf_to_utc_timestamp.q
@@ -1,24 +1,24 @@
 DESCRIBE FUNCTION to_utc_timestamp;
 DESC FUNCTION EXTENDED to_utc_timestamp;
 
-explain select to_utc_timestamp('2015-02-11 10:30:00', 'PST');
+explain select to_utc_timestamp('2012-02-11 10:30:00', 'PST');
 
 select
-to_utc_timestamp('2015-02-10 20:30:00', 'PST'),
-to_utc_timestamp('2015-02-11 08:30:00', 'Europe/Moscow'),
-to_utc_timestamp('2015-02-11 12:30:00', 'GMT+8'),
-to_utc_timestamp('2015-02-11 04:30:00', 'GMT'),
-to_utc_timestamp('2015-02-11 04:30:00', ''),
-to_utc_timestamp('2015-02-11 04:30:00', '---'),
+to_utc_timestamp('2012-02-10 20:30:00', 'PST'),
+to_utc_timestamp('2012-02-11 08:30:00', 'Europe/Moscow'),
+to_utc_timestamp('2012-02-11 12:30:00', 'GMT+8'),
+to_utc_timestamp('2012-02-11 04:30:00', 'GMT'),
+to_utc_timestamp('2012-02-11 04:30:00', ''),
+to_utc_timestamp('2012-02-11 04:30:00', '---'),
 to_utc_timestamp(cast(null as string), 'PST'),
-to_utc_timestamp('2015-02-11 04:30:00', cast(null as string));
+to_utc_timestamp('2012-02-11 04:30:00', cast(null as string));
 
 select
-to_utc_timestamp(cast('2015-02-10 20:30:00' as timestamp), 'PST'),
-to_utc_timestamp(cast('2015-02-11 08:30:00' as timestamp), 'Europe/Moscow'),
-to_utc_timestamp(cast('2015-02-11 12:30:00' as timestamp), 'GMT+8'),
-to_utc_timestamp(cast('2015-02-11 04:30:00' as timestamp), 'GMT'),
-to_utc_timestamp(cast('2015-02-11 04:30:00' as timestamp), ''),
-to_utc_timestamp(cast('2015-02-11 04:30:00' as timestamp), '---'),
+to_utc_timestamp(cast('2012-02-10 20:30:00' as timestamp), 'PST'),
+to_utc_timestamp(cast('2012-02-11 08:30:00' as timestamp), 'Europe/Moscow'),
+to_utc_timestamp(cast('2012-02-11 12:30:00' as timestamp), 'GMT+8'),
+to_utc_timestamp(cast('2012-02-11 04:30:00' as timestamp), 'GMT'),
+to_utc_timestamp(cast('2012-02-11 04:30:00' as timestamp), ''),
+to_utc_timestamp(cast('2012-02-11 04:30:00' as timestamp), '---'),
 to_utc_timestamp(cast(null as timestamp), 'PST'),
-to_utc_timestamp(cast('2015-02-11 04:30:00' as timestamp), cast(null as string));
\ No newline at end of file
+to_utc_timestamp(cast('2012-02-11 04:30:00' as timestamp), cast(null as string));
diff --git a/ql/src/test/results/clientpositive/udf_from_utc_timestamp.q.out b/ql/src/test/results/clientpositive/udf_from_utc_timestamp.q.out
index 4df872c..001d42c 100644
--- a/ql/src/test/results/clientpositive/udf_from_utc_timestamp.q.out
+++ b/ql/src/test/results/clientpositive/udf_from_utc_timestamp.q.out
@@ -8,9 +8,9 @@ PREHOOK: type: DESCFUNCTION
 POSTHOOK: query: DESC FUNCTION EXTENDED from_utc_timestamp
 POSTHOOK: type: DESCFUNCTION
 from_utc_timestamp(timestamp, string timezone) - Assumes given timestamp is UTC and converts to given timezone (as of Hive 0.8.0)
-PREHOOK: query: explain select from_utc_timestamp('2015-02-11 10:30:00', 'PST')
+PREHOOK: query: explain select from_utc_timestamp('2012-02-11 10:30:00', 'PST')
 PREHOOK: type: QUERY
-POSTHOOK: query: explain select from_utc_timestamp('2015-02-11 10:30:00', 'PST')
+POSTHOOK: query: explain select from_utc_timestamp('2012-02-11 10:30:00', 'PST')
 POSTHOOK: type: QUERY
 STAGE DEPENDENCIES:
   Stage-0 is a root stage
@@ -25,58 +25,58 @@ STAGE PLANS:
           Row Limit Per Split: 1
           Statistics: Num rows: 0 Data size: 1 Basic stats: PARTIAL Column stats: COMPLETE
           Select Operator
-            expressions: 2015-02-11 02:30:00.0 (type: timestamp)
+            expressions: 2012-02-11 02:30:00.0 (type: timestamp)
             outputColumnNames: _col0
             Statistics: Num rows: 0 Data size: 1 Basic stats: PARTIAL Column stats: COMPLETE
             ListSink
 
 PREHOOK: query: select
-from_utc_timestamp('2015-02-11 04:30:00', 'PST'),
-from_utc_timestamp('2015-02-11 04:30:00', 'Europe/Moscow'),
-from_utc_timestamp('2015-02-11 04:30:00', 'GMT+8'),
-from_utc_timestamp('2015-02-11 04:30:00', 'GMT'),
-from_utc_timestamp('2015-02-11 04:30:00', ''),
-from_utc_timestamp('2015-02-11 04:30:00', '---'),
+from_utc_timestamp('2012-02-11 04:30:00', 'PST'),
+from_utc_timestamp('2012-02-11 04:30:00', 'Europe/Moscow'),
+from_utc_timestamp('2012-02-11 04:30:00', 'GMT+8'),
+from_utc_timestamp('2012-02-11 04:30:00', 'GMT'),
+from_utc_timestamp('2012-02-11 04:30:00', ''),
+from_utc_timestamp('2012-02-11 04:30:00', '---'),
 from_utc_timestamp(cast(null as string), 'PST'),
-from_utc_timestamp('2015-02-11 04:30:00', cast(null as string))
+from_utc_timestamp('2012-02-11 04:30:00', cast(null as string))
 PREHOOK: type: QUERY
 PREHOOK: Input: _dummy_database@_dummy_table
 #### A masked pattern was here ####
 POSTHOOK: query: select
-from_utc_timestamp('2015-02-11 04:30:00', 'PST'),
-from_utc_timestamp('2015-02-11 04:30:00', 'Europe/Moscow'),
-from_utc_timestamp('2015-02-11 04:30:00', 'GMT+8'),
-from_utc_timestamp('2015-02-11 04:30:00', 'GMT'),
-from_utc_timestamp('2015-02-11 04:30:00', ''),
-from_utc_timestamp('2015-02-11 04:30:00', '---'),
+from_utc_timestamp('2012-02-11 04:30:00', 'PST'),
+from_utc_timestamp('2012-02-11 04:30:00', 'Europe/Moscow'),
+from_utc_timestamp('2012-02-11 04:30:00', 'GMT+8'),
+from_utc_timestamp('2012-02-11 04:30:00', 'GMT'),
+from_utc_timestamp('2012-02-11 04:30:00', ''),
+from_utc_timestamp('2012-02-11 04:30:00', '---'),
 from_utc_timestamp(cast(null as string), 'PST'),
-from_utc_timestamp('2015-02-11 04:30:00', cast(null as string))
+from_utc_timestamp('2012-02-11 04:30:00', cast(null as string))
 POSTHOOK: type: QUERY
 POSTHOOK: Input: _dummy_database@_dummy_table
 #### A masked pattern was here ####
-2015-02-10 20:30:00	2015-02-11 08:30:00	2015-02-11 12:30:00	2015-02-11 04:30:00	2015-02-11 04:30:00	2015-02-11 04:30:00	NULL	NULL
+2012-02-10 20:30:00	2012-02-11 08:30:00	2012-02-11 12:30:00	2012-02-11 04:30:00	2012-02-11 04:30:00	2012-02-11 04:30:00	NULL	NULL
 PREHOOK: query: select
-from_utc_timestamp(cast('2015-02-11 04:30:00' as timestamp), 'PST'),
-from_utc_timestamp(cast('2015-02-11 04:30:00' as timestamp), 'Europe/Moscow'),
-from_utc_timestamp(cast('2015-02-11 04:30:00' as timestamp), 'GMT+8'),
-from_utc_timestamp(cast('2015-02-11 04:30:00' as timestamp), 'GMT'),
-from_utc_timestamp(cast('2015-02-11 04:30:00' as timestamp), ''),
-from_utc_timestamp(cast('2015-02-11 04:30:00' as timestamp), '---'),
+from_utc_timestamp(cast('2012-02-11 04:30:00' as timestamp), 'PST'),
+from_utc_timestamp(cast('2012-02-11 04:30:00' as timestamp), 'Europe/Moscow'),
+from_utc_timestamp(cast('2012-02-11 04:30:00' as timestamp), 'GMT+8'),
+from_utc_timestamp(cast('2012-02-11 04:30:00' as timestamp), 'GMT'),
+from_utc_timestamp(cast('2012-02-11 04:30:00' as timestamp), ''),
+from_utc_timestamp(cast('2012-02-11 04:30:00' as timestamp), '---'),
 from_utc_timestamp(cast(null as timestamp), 'PST'),
-from_utc_timestamp(cast('2015-02-11 04:30:00' as timestamp), cast(null as string))
+from_utc_timestamp(cast('2012-02-11 04:30:00' as timestamp), cast(null as string))
 PREHOOK: type: QUERY
 PREHOOK: Input: _dummy_database@_dummy_table
 #### A masked pattern was here ####
 POSTHOOK: query: select
-from_utc_timestamp(cast('2015-02-11 04:30:00' as timestamp), 'PST'),
-from_utc_timestamp(cast('2015-02-11 04:30:00' as timestamp), 'Europe/Moscow'),
-from_utc_timestamp(cast('2015-02-11 04:30:00' as timestamp), 'GMT+8'),
-from_utc_timestamp(cast('2015-02-11 04:30:00' as timestamp), 'GMT'),
-from_utc_timestamp(cast('2015-02-11 04:30:00' as timestamp), ''),
-from_utc_timestamp(cast('2015-02-11 04:30:00' as timestamp), '---'),
+from_utc_timestamp(cast('2012-02-11 04:30:00' as timestamp), 'PST'),
+from_utc_timestamp(cast('2012-02-11 04:30:00' as timestamp), 'Europe/Moscow'),
+from_utc_timestamp(cast('2012-02-11 04:30:00' as timestamp), 'GMT+8'),
+from_utc_timestamp(cast('2012-02-11 04:30:00' as timestamp), 'GMT'),
+from_utc_timestamp(cast('2012-02-11 04:30:00' as timestamp), ''),
+from_utc_timestamp(cast('2012-02-11 04:30:00' as timestamp), '---'),
 from_utc_timestamp(cast(null as timestamp), 'PST'),
-from_utc_timestamp(cast('2015-02-11 04:30:00' as timestamp), cast(null as string))
+from_utc_timestamp(cast('2012-02-11 04:30:00' as timestamp), cast(null as string))
 POSTHOOK: type: QUERY
 POSTHOOK: Input: _dummy_database@_dummy_table
 #### A masked pattern was here ####
-2015-02-10 20:30:00	2015-02-11 08:30:00	2015-02-11 12:30:00	2015-02-11 04:30:00	2015-02-11 04:30:00	2015-02-11 04:30:00	NULL	NULL
+2012-02-10 20:30:00	2012-02-11 08:30:00	2012-02-11 12:30:00	2012-02-11 04:30:00	2012-02-11 04:30:00	2012-02-11 04:30:00	NULL	NULL
diff --git a/ql/src/test/results/clientpositive/udf_to_utc_timestamp.q.out b/ql/src/test/results/clientpositive/udf_to_utc_timestamp.q.out
index e70b401..84fe118 100644
--- a/ql/src/test/results/clientpositive/udf_to_utc_timestamp.q.out
+++ b/ql/src/test/results/clientpositive/udf_to_utc_timestamp.q.out
@@ -8,9 +8,9 @@ PREHOOK: type: DESCFUNCTION
 POSTHOOK: query: DESC FUNCTION EXTENDED to_utc_timestamp
 POSTHOOK: type: DESCFUNCTION
 to_utc_timestamp(timestamp, string timezone) - Assumes given timestamp is in given timezone and converts to UTC (as of Hive 0.8.0)
-PREHOOK: query: explain select to_utc_timestamp('2015-02-11 10:30:00', 'PST')
+PREHOOK: query: explain select to_utc_timestamp('2012-02-11 10:30:00', 'PST')
 PREHOOK: type: QUERY
-POSTHOOK: query: explain select to_utc_timestamp('2015-02-11 10:30:00', 'PST')
+POSTHOOK: query: explain select to_utc_timestamp('2012-02-11 10:30:00', 'PST')
 POSTHOOK: type: QUERY
 STAGE DEPENDENCIES:
   Stage-0 is a root stage
@@ -25,58 +25,58 @@ STAGE PLANS:
           Row Limit Per Split: 1
           Statistics: Num rows: 0 Data size: 1 Basic stats: PARTIAL Column stats: COMPLETE
           Select Operator
-            expressions: 2015-02-11 18:30:00.0 (type: timestamp)
+            expressions: 2012-02-11 18:30:00.0 (type: timestamp)
             outputColumnNames: _col0
             Statistics: Num rows: 0 Data size: 1 Basic stats: PARTIAL Column stats: COMPLETE
             ListSink
 
 PREHOOK: query: select
-to_utc_timestamp('2015-02-10 20:30:00', 'PST'),
-to_utc_timestamp('2015-02-11 08:30:00', 'Europe/Moscow'),
-to_utc_timestamp('2015-02-11 12:30:00', 'GMT+8'),
-to_utc_timestamp('2015-02-11 04:30:00', 'GMT'),
-to_utc_timestamp('2015-02-11 04:30:00', ''),
-to_utc_timestamp('2015-02-11 04:30:00', '---'),
+to_utc_timestamp('2012-02-10 20:30:00', 'PST'),
+to_utc_timestamp('2012-02-11 08:30:00', 'Europe/Moscow'),
+to_utc_timestamp('2012-02-11 12:30:00', 'GMT+8'),
+to_utc_timestamp('2012-02-11 04:30:00', 'GMT'),
+to_utc_timestamp('2012-02-11 04:30:00', ''),
+to_utc_timestamp('2012-02-11 04:30:00', '---'),
 to_utc_timestamp(cast(null as string), 'PST'),
-to_utc_timestamp('2015-02-11 04:30:00', cast(null as string))
+to_utc_timestamp('2012-02-11 04:30:00', cast(null as string))
 PREHOOK: type: QUERY
 PREHOOK: Input: _dummy_database@_dummy_table
 #### A masked pattern was here ####
 POSTHOOK: query: select
-to_utc_timestamp('2015-02-10 20:30:00', 'PST'),
-to_utc_timestamp('2015-02-11 08:30:00', 'Europe/Moscow'),
-to_utc_timestamp('2015-02-11 12:30:00', 'GMT+8'),
-to_utc_timestamp('2015-02-11 04:30:00', 'GMT'),
-to_utc_timestamp('2015-02-11 04:30:00', ''),
-to_utc_timestamp('2015-02-11 04:30:00', '---'),
+to_utc_timestamp('2012-02-10 20:30:00', 'PST'),
+to_utc_timestamp('2012-02-11 08:30:00', 'Europe/Moscow'),
+to_utc_timestamp('2012-02-11 12:30:00', 'GMT+8'),
+to_utc_timestamp('2012-02-11 04:30:00', 'GMT'),
+to_utc_timestamp('2012-02-11 04:30:00', ''),
+to_utc_timestamp('2012-02-11 04:30:00', '---'),
 to_utc_timestamp(cast(null as string), 'PST'),
-to_utc_timestamp('2015-02-11 04:30:00', cast(null as string))
+to_utc_timestamp('2012-02-11 04:30:00', cast(null as string))
 POSTHOOK: type: QUERY
 POSTHOOK: Input: _dummy_database@_dummy_table
 #### A masked pattern was here ####
-2015-02-11 04:30:00	2015-02-11 04:30:00	2015-02-11 04:30:00	2015-02-11 04:30:00	2015-02-11 04:30:00	2015-02-11 04:30:00	NULL	NULL
+2012-02-11 04:30:00	2012-02-11 04:30:00	2012-02-11 04:30:00	2012-02-11 04:30:00	2012-02-11 04:30:00	2012-02-11 04:30:00	NULL	NULL
 PREHOOK: query: select
-to_utc_timestamp(cast('2015-02-10 20:30:00' as timestamp), 'PST'),
-to_utc_timestamp(cast('2015-02-11 08:30:00' as timestamp), 'Europe/Moscow'),
-to_utc_timestamp(cast('2015-02-11 12:30:00' as timestamp), 'GMT+8'),
-to_utc_timestamp(cast('2015-02-11 04:30:00' as timestamp), 'GMT'),
-to_utc_timestamp(cast('2015-02-11 04:30:00' as timestamp), ''),
-to_utc_timestamp(cast('2015-02-11 04:30:00' as timestamp), '---'),
+to_utc_timestamp(cast('2012-02-10 20:30:00' as timestamp), 'PST'),
+to_utc_timestamp(cast('2012-02-11 08:30:00' as timestamp), 'Europe/Moscow'),
+to_utc_timestamp(cast('2012-02-11 12:30:00' as timestamp), 'GMT+8'),
+to_utc_timestamp(cast('2012-02-11 04:30:00' as timestamp), 'GMT'),
+to_utc_timestamp(cast('2012-02-11 04:30:00' as timestamp), ''),
+to_utc_timestamp(cast('2012-02-11 04:30:00' as timestamp), '---'),
 to_utc_timestamp(cast(null as timestamp), 'PST'),
-to_utc_timestamp(cast('2015-02-11 04:30:00' as timestamp), cast(null as string))
+to_utc_timestamp(cast('2012-02-11 04:30:00' as timestamp), cast(null as string))
 PREHOOK: type: QUERY
 PREHOOK: Input: _dummy_database@_dummy_table
 #### A masked pattern was here ####
 POSTHOOK: query: select
-to_utc_timestamp(cast('2015-02-10 20:30:00' as timestamp), 'PST'),
-to_utc_timestamp(cast('2015-02-11 08:30:00' as timestamp), 'Europe/Moscow'),
-to_utc_timestamp(cast('2015-02-11 12:30:00' as timestamp), 'GMT+8'),
-to_utc_timestamp(cast('2015-02-11 04:30:00' as timestamp), 'GMT'),
-to_utc_timestamp(cast('2015-02-11 04:30:00' as timestamp), ''),
-to_utc_timestamp(cast('2015-02-11 04:30:00' as timestamp), '---'),
+to_utc_timestamp(cast('2012-02-10 20:30:00' as timestamp), 'PST'),
+to_utc_timestamp(cast('2012-02-11 08:30:00' as timestamp), 'Europe/Moscow'),
+to_utc_timestamp(cast('2012-02-11 12:30:00' as timestamp), 'GMT+8'),
+to_utc_timestamp(cast('2012-02-11 04:30:00' as timestamp), 'GMT'),
+to_utc_timestamp(cast('2012-02-11 04:30:00' as timestamp), ''),
+to_utc_timestamp(cast('2012-02-11 04:30:00' as timestamp), '---'),
 to_utc_timestamp(cast(null as timestamp), 'PST'),
-to_utc_timestamp(cast('2015-02-11 04:30:00' as timestamp), cast(null as string))
+to_utc_timestamp(cast('2012-02-11 04:30:00' as timestamp), cast(null as string))
 POSTHOOK: type: QUERY
 POSTHOOK: Input: _dummy_database@_dummy_table
 #### A masked pattern was here ####
-2015-02-11 04:30:00	2015-02-11 04:30:00	2015-02-11 04:30:00	2015-02-11 04:30:00	2015-02-11 04:30:00	2015-02-11 04:30:00	NULL	NULL
+2012-02-11 04:30:00	2012-02-11 04:30:00	2012-02-11 04:30:00	2012-02-11 04:30:00	2012-02-11 04:30:00	2012-02-11 04:30:00	NULL	NULL
-- 
1.7.9.5

